src/pager.c

```diff
@@ -1429,6 +1430,18 @@ char *supplemental_name;
         if (*dbase_str == ' ')
             ++dbase_str;
     }
+
+    /* remove material type */
+    for (i = 1; i < NUM_MATERIAL_TYPES; i++) {
+        l = strlen(materialnm[i]);
+        Strcpy(buf, materialnm[i]);
+        Strcat(buf, " ");
```

Comment:
I'm not super familiar with all the NetHack string manipulation functions (`Strcat`, etc), so I'm not sure if this is the best or most standard way to turn `"gold"` into `"gold "` for comparison to `dbase_str` (to ensure we aren't turning `"golden gun"` into `"en gun"`) -- is there a better way to do it?

Reply:
This could always be done with 
```c
strncmp(dbase_str, materialnm[i], l) && strlen(dbase_str) > l && dbase_str[l] == ' '
```
which avoids using a secondary buffer at all.